contracts/Tokens/Prime/Prime.sol

```diff
@@ -615,11 +634,53 @@ contract Prime is IIncomeDestination, AccessControlledV8, PrimeStorageV1 {
         }
     }
 
-    function _getUnderlying(address vToken) internal view returns (address) {
-        if (vToken == vBNB) {
-            return WBNB;
-        } else {
-            return IVToken(vToken).underlying();
-        }
+    //////////////////////////////////////////////////
+    //////////////// APR Calculation ////////////////
+    ////////////////////////////////////////////////    
+
+    function _incomePerBlock(address vToken) internal view returns (uint256) {
+        IVToken market = IVToken(vToken);
+        return ((((market.totalBorrows() * market.borrowRatePerBlock()) / EXP_SCALE) * market.reserveFactorMantissa()) / EXP_SCALE);
+    }
+
+    function _distributionPercentage() internal view returns (uint256) {
+        return IProtocolShareReserve(protocolShareReserve).getPercentageDistribution(
+            address(this), IProtocolShareReserve.Schema.SPREAD_PRIME_CORE
+        );
+    }
+
+    function _incomeDistributionYearly(address vToken) internal view returns (uint256) {
+        uint256 totalIncomePerBlock = _incomePerBlock(vToken);
+        uint256 incomePerBlockForDistribution = (totalIncomePerBlock * _distributionPercentage()) / IProtocolShareReserve(protocolShareReserve).MAX_PERCENT();
+        return BLOCKS_PER_YEAR * incomePerBlockForDistribution;
+    }
+
+    function _calculateUserAPR(
+        address vToken, 
+        address user, 
+        uint256 totalSupply, 
+        uint256 totalBorrow
+    ) internal view returns (uint256 supplyAPR, uint256 borrowAPR) {
+        uint256 userScore = interests[vToken][user].score;
+        uint256 totalScore = markets[vToken].sumOfMembersScore;
+
+        uint256 userYearlyIncome = (userScore * _incomeDistributionYearly(vToken)) / totalScore;
+        uint256 totalValue = totalSupply + totalBorrow;
+
+        uint256 userSupplyIncomeYearly = (userYearlyIncome * totalSupply) / totalValue;
+        uint256 userBorrowIncomeYearly = (userYearlyIncome * totalBorrow) / totalValue;
+
+        supplyAPR = (userSupplyIncomeYearly * MAXIMUM_BPS) / totalSupply;
+        borrowAPR = (userBorrowIncomeYearly * MAXIMUM_BPS) / totalBorrow;
+    }
+
+    function apr(address market, address user) external view returns (uint256 supplyAPR, uint256 borrowAPR) {
+        IVToken vToken = IVToken(market);
+        uint256 borrow = vToken.borrowBalanceStored(user);
+        uint256 exchangeRate = vToken.exchangeRateStored();
+        uint256 balanceOfAccount = vToken.balanceOf(user);
+        uint256 supply = (exchangeRate * balanceOfAccount) / EXP_SCALE;
+
+        return _calculateUserAPR(market, user, supply, borrow);
```

Comment:
If we provided the user `score` to this function, we'll be able to reuse it for the `estimatedAPR` function, where we would need to calculate the score before invoking `_calculateUserAPR`